src/main/java/sg/ncl/MainController.java

```diff
@@ -1880,18 +1891,28 @@ public String stopExperiment(@PathVariable String teamName, @PathVariable String
         try {
             if (RestUtil.isError(response.getStatusCode())) {
                 MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
-                return "redirect:/experiments";
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == ExceptionState.FORBIDDEN_EXCEPTION) {
+                    log.warn("Permission denied to stop experiment: {} for team: {}", realization.getExperimentName(), teamName);
+                    redirectAttributes.addFlashAttribute("message", permissionDeniedMessage);
+                }
             } else {
                 // everything ok
                 log.info("stop experiment success for Team: {}, Exp: {}", teamName, expId);
                 redirectAttributes.addFlashAttribute("exp_message", "Team: " + teamName + " has stopped Exp: " + realization.getExperimentName());
-                return "redirect:/experiments";
             }
+            return "redirect:/experiments";
         } catch (IOException e) {
             throw new WebServiceRuntimeException(e.getMessage());
         }
     }
 
+    private boolean isNotAdminAndNotInTeam(HttpSession session, Realization realization) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "session". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)